server/controllers/users.js

```diff
@@ -72,36 +72,36 @@ export const signin = (req, res, next) => {
 };
 
 /**
- * Receive a JWT and generate another one.
+ * Receive a login JWT and generate another one.
  * This can be used right after the first login.
- * If an authenticator code is sent, check if
- * the user has two-factor authentication enabled
- * on their account, and if they do, we challenge
- * them to auth with it during the login flow
+ * Also check if the user has two-factor authentication
+ * enabled on their account, and if they do, we send
+ * back a JWT with scope 'twofactorauth' to trigger
+ * the 2FA flow on the frontend
  */
-export const updateToken = async (req, res, next) => {
-  const { twoFactorAuthenticatorCode } = req.body;
-
-  // the first time we try, we need to just check if the user has 2FA
-  if (req.remoteUser.twoFactorAuthToken !== null && !twoFactorAuthenticatorCode) {
-    return next(new Unauthorized('Two-factor authentication is enabled on this account. Please enter the code'));
+export const updateToken = async (req, res) => {
+  if (req.remoteUser.twoFactorAuthToken !== null) {
+    const token = req.remoteUser.jwt({ scope: 'twofactorauth' }, auth.TOKEN_EXPIRATION_SESSION);
```

Comment:
Seems to me this token has no limitation and can be used as a regular token.

If true, this defeats the 2FA and allow anyone with this token to perform actions without being confirmed.